### PR TITLE
Ingest role bindings

### LINKED ISSUE #207: Ingest Role Bindings

### PR DESCRIPTION
**Pull request by mknoedel**
fixes #207 

- Added support for ingesting the following **new** resources:

  | Service     | Resource / Entity    |
  | ----------- | -------------------- |
  | IAM Binding | `google_iam_binding` |

This allows the query:

```
find google_iam_binding with members=(
  "allUsers" or
  "allAuthenticatedUsers"
) (THAT ASSIGNED * as resource)?
return TREE
/*
  google_iam_binding.resource
*/
```

There are no relationships made in this PR, but they will be added in later PRs.

### CODE REVIEW

File: src/steps/cloud-asset/constants.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      bindingEntities.BINDINGS._type,
+      targetType,
+    ),
+    sourceType: bindingEntities.BINDINGS._type,
+    _class: RelationshipClass.ASSIGNED,
+    targetType,
+  };
+});
+
+export const bindingRelationships = {
```
Comment (mknoedel) [inline src/steps/cloud-asset/constants.ts:39]: These are not yet used. The actual relationships will be created in later PRs.

File: src/steps/cloud-asset/index.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    for (const binding of bindings) {
+      await jobState.addEntity(
+        createIamBindingEntity(binding, project, resource),
+      );
+      iamBindingsCount++;
+    }
+  });
+
+  logger.info(
+    { numIamBindings: iamBindingsCount },
```
Comment (austinkelleher) [inline src/steps/cloud-asset/index.ts:29]: 👍

File: src/steps/cloud-asset/converters.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  binding: cloudasset_v1.Schema$Binding,
+  project: string | undefined | null,
+  resource: string | undefined | null,
+) {
+  // There are no unique values returned from this result so use everything available to make the binding key unique.
+  const keyBuilders = ['binding'];
+  if (project) keyBuilders.push('project:' + project);
+  if (resource) keyBuilders.push('resource:' + resource);
+  if (binding.role) keyBuilders.push('role:' + binding.role);
+  const key = keyBuilders.join('|');
```
Comment (austinkelleher) [inline src/steps/cloud-asset/converters.ts:17]: Nice. This key should be quite unique.

**Review by austinkelleher — APPROVED** [review]

File: src/steps/cloud-asset/converters.ts (hunk 3 above)
Comment (ndowmon) [reply src/steps/cloud-asset/converters.ts:17]: Does this composite _also_ need the principal (user, group, etc) in order to be unique? @austinkelleher @mknoedel

Comment (ndowmon) [reply src/steps/cloud-asset/converters.ts:17]: Ah I see that this can relate to _many_ principals via the `members` property. Disregard!

**Pull request merged**